src/validate.js

```diff
@@ -2429,30 +2429,31 @@ function F_VALIDATE(disabledHL) {
 
 			// check Public connection
 			if (slowChecks
-				&& segment.$isRoutable){
+				&& segment.$isRoutable
+				&& nodeA.$otherSegments.length > 0
+				&& nodeB.$otherSegments.length > 0){
 				// Check other segments to be a drivable public segment
 				var foundPublicConnection = false;
-				for (var i = 0; i < nodeA.$otherSegmentsLen; i++) {
+				for (var i = 0; i < nodeA.$otherSegments.length; i++) {
```

Comment:
`$otherSegments.length` should have the same value as `$otherSegmentsLen`, but will be a bit slower due to the `$otherSegments` getter described above.